### PR TITLE
refactor: cleanup interfaces + add missing methods from parser-api

### PR DESCRIPTION
**Pull request by smoya**
**Description**

This PR does a cleanup and sync with the latest changes on `parser-api` made in https://github.com/asyncapi/parser-api/pull/71#issuecomment-1239379887.

**Related issue(s)**
- https://github.com/asyncapi/parser-js/issues/481
- https://github.com/asyncapi/parser-js/issues/482

### CODE REVIEW

File: src/models/security-scheme.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import type { OAuthFlowsInterface } from './oauth-flows';
 import type { DescriptionMixinInterface, ExtensionsMixinInterface } from './mixins';
 
 export type SecuritySchemaType = 'userPassword' | 'apiKey' | 'X509' | 'symmetricEncryption' | 'asymmetricEncryption' | 'httpApiKey' | 'http' | 'oauth2' | 'openIdConnect' | 'plain' | 'scramSha256' | 'scramSha512' | 'gssapi';
+export type SecuritySchemaLocation = 'user' | 'password' | 'query' | 'header' | 'header' | 'cookie';
```
Comment (magicmatatjahu) [inline src/models/security-scheme.ts:6]: Hmm 🤔 Probably we should move that types to the `spec-types/v2`. `SecuritySchemaType` probably I wrote but we have missed `SecuritySchemaLocation` here. Could you move them?

File: src/models/v2/extensions.ts [hunk 2]
```diff
@@ -6,6 +6,6 @@ import type { ExtensionInterface } from '../extension';
 export class Extensions extends Collection<ExtensionInterface> implements ExtensionsInterface {
   override get<T = any>(name: string): ExtensionInterface<T> | undefined {
     name = name.startsWith('x-') ? name : `x-${name}`;
```
Comment (magicmatatjahu) [inline src/models/v2/extensions.ts:8]:
```suggestion
  override get<T = any>(id: string): ExtensionInterface<T> | undefined {
    id = id.startsWith('x-') ? id : `x-${id}`;
```

File: src/models/v2/extensions.ts [hunk 3]
```diff
@@ -6,6 +6,6 @@ import type { ExtensionInterface } from '../extension';
...
-    return this.collections.find(ext => ext.name() === name);
+    return this.collections.find(ext => ext.id() === name);
```
Comment (magicmatatjahu) [inline src/models/v2/extensions.ts:9]:
```suggestion
    return this.collections.find(ext => ext.id() === id);
```

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]

File: src/models/security-scheme.ts (hunk 1 above)
Comment (smoya) [reply src/models/security-scheme.ts:6]: Those are already in `spec-types/v2`. The question is, should we also add **all** of them (meaning all locations possible, no matters the version of the spec) in the interface?That was the idea behind this in `parser-api`: https://github.com/asyncapi/parser-api/pull/71/files#diff-ac35ac6e0b0ad3629802a0e04076aa18e94da614f5392e1164dcaacdfe3ee198R375.
WDYT?

Comment (magicmatatjahu) [reply src/models/security-scheme.ts:6]: Yeah, we don't make different spec types for `2.0.0`, `2.1.0` etc. v2 spec types is for 2.x.x, so please use types from `spec-types/v2`.

Comment (smoya) [reply src/models/security-scheme.ts:6]: > v2 spec types is for 2.x.x, so please use types from spec-types/v2

But we are talking about types located in the interface (`src/models/security-scheme.ts`). Not related with the implementation.

Comment (smoya) [reply src/models/security-scheme.ts:6]: Do you mean we should just use `string` as type here, and in the implementation (v2, v3...) to set those values as unique for that string?

Comment (magicmatatjahu) [reply src/models/security-scheme.ts:6]: We can even use `v2.SecuritySchemaLocation | v3.SecuritySchemaLocation` type in the future, but probably better will be `string` and then in implementation cast type to the `v2.SecuritySchemaLocation`. Please change to the `string` and move types to the `spec-types/v2`

Comment (smoya) [reply src/models/security-scheme.ts]: Done!

**Review by magicmatatjahu — APPROVED** [review]
LGTM!

**Pull request merged**